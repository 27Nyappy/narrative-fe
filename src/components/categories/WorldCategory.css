.WorldCategory {
  background-color: aqua;
  width: 50%;
  display: flex;
  flex-direction: column;
  justify-content: center;

  h2, div {
    align-self: center;
  }
}

.WorldCategoryContainer {
  width: 85%;
  height: 40%;
  display: flex;
  justify-content: space-between;
}