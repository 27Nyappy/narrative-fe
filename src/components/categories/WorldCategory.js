import React from 'react';
import PropTypes from 'prop-types';
import AddWorldButton from './AddWorldButton';
import ViewAllButton from './ViewAllButton';
import styles from './WorldCategory.css';

const WorldCategory = ({ handleViewWorlds, show, hideModal, showModal }) => {
  return (
    <section className={styles.WorldCategory}>
      <h2 className={styles.h2}>World Building</h2>

      <div className={styles.div}>
        <AddWorldButton show={show} hideModal={hideModal} showModal={showModal} />
        <ViewAllButton handleViewClick={handleViewWorlds} />
      </div>
    </section>
  );
};

WorldCategory.propTypes = {
  handleViewWorlds: PropTypes.func.isRequired,
  show: PropTypes.bool,
  hideModal: PropTypes.func,
  showModal: PropTypes.func
};

export default WorldCategory;