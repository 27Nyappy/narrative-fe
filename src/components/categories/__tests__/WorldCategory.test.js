import React from 'react';
import { shallow } from 'enzyme';
import WorldCategory from '../WorldCategory';

describe('WorldCategory component', () => {
  it('renders a world category', () => {
    const wrapper = shallow(<WorldCategory handleAddWorld={() => true} handleViewWorlds={() => true}/>);
    expect(wrapper).toMatchSnapshot(); 
  });
});