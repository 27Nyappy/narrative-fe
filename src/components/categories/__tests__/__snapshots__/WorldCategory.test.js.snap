// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorldCategory component renders a world category 1`] = `
<section>
  <h2>
    World Building
  </h2>
  <div>
    <AddButton
      handleAddClick={[Function]}
    />
    <ViewAllButton
      handleViewClick={[Function]}
    />
  </div>
</section>
`;
